service/controller/v18/resource/ipam/create.go

```diff
@@ -104,13 +128,29 @@ func (r *Resource) allocateSubnet(ctx context.Context) (string, error) {
 
 		subnet, err = ipam.Free(r.networkRange, r.allocatedSubnetMask, reservedSubnets)
 		if err != nil {
-			return "", microerror.Maskf(err, "networkRange: %s, allocatedSubnetMask: %s, reservedSubnets: %#v", r.networkRange.String(), r.allocatedSubnetMask.String(), reservedSubnets)
+			return net.IPNet{}, microerror.Maskf(err, "networkRange: %s, allocatedSubnetMask: %s, reservedSubnets: %#v", r.networkRange.String(), r.allocatedSubnetMask.String(), reservedSubnets)
 		}
 
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found free subnet %#q", subnet.String()))
 	}
 
-	return subnet.String(), nil
+	return subnet, nil
+}
+
+func (r *Resource) selectRandomAZs(n int) ([]string, error) {
+	if n > len(r.availabilityZones) {
+		return nil, microerror.Maskf(invalidParameterError, "requested nubmer of AZs %d is bigger than number of available AZs %d", n, len(r.availabilityZones))
+	}
```

Comment:
This check should be done during the resource initialization. I thought about having these information given via the call parameters. Then this would make sense, but the internals should be taken for granted during runtime at this point.

Reply:
I don't understand what you mean here. Can you elaborate?

Reply:
IMO we can remove this check here because the resource initialization should verify this.

Reply:
No, this check here verifies that given number of requested AZs in CR object is not bigger than number of configured AZs in operator. We don't know CR object values during resource initialization.